admin/plugins.php

```diff
@@ -0,0 +1,399 @@
+						}
+						?>
+					</div>
+				</div>
+			</form>
+		</div>
+		<div class="clear"></div>
+	</div>
+	<?php
+}
+
+/**
+ * Render individual plugin cards.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $standalone_plugin Plugin slug as passed from get_standalone_plugins().
+ */
+function perflab_render_plugin_card( string $standalone_plugin ) {
+	$plugin = plugins_api(
+		'plugin_information',
+		array(
+			'slug'   => $standalone_plugin,
+			'fields' => array(
+				'short_description' => true,
+				'icons'             => true,
+			),
+		)
+	);
+	if ( is_object( $plugin ) ) {
+		$plugin = (array) $plugin;
+	}
+
+	// Remove any HTML from the description.
+	$description = wp_strip_all_tags( $plugin['short_description'] );
+	$title       = $plugin['name'];
+
+	/**
+	 * Filters the plugin card description on the Add Plugins screen.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $description Plugin card description.
+	 * @param array  $plugin      An array of plugin data. See {@see plugins_api()}
+	 *                            for the list of possible values.
+	 */
+	$description = apply_filters( 'plugin_install_description', $description, $plugin );
+	$version     = $plugin['version'];
+	$name        = wp_strip_all_tags( $title . ' ' . $version );
+	$author      = $plugin['author'];
+	if ( ! empty( $author ) ) {
+		/* translators: %s: Plugin author. */
+		$author = ' <cite>' . sprintf( __( 'By %s', 'default' ), $author ) . '</cite>';
+	}
+
+	$requires_php = isset( $plugin['requires_php'] ) ? $plugin['requires_php'] : null;
+	$requires_wp  = isset( $plugin['requires'] ) ? $plugin['requires'] : null;
+
+	$compatible_php = is_php_version_compatible( $requires_php );
+	$compatible_wp  = is_wp_version_compatible( $requires_wp );
+	$tested_wp      = ( empty( $plugin['tested'] ) || version_compare( get_bloginfo( 'version' ), $plugin['tested'], '<=' ) );
+	$action_links   = array();
+
+	if ( current_user_can( 'install_plugins' ) || current_user_can( 'update_plugins' ) ) {
+		$status = install_plugin_install_status( $plugin );
+
+		switch ( $status['status'] ) {
+			case 'install':
+				if ( $status['url'] ) {
+					if ( $compatible_php && $compatible_wp ) {
+						$action_links[] = sprintf(
+							'<a class="install-now button" data-slug="%s" href="%s" aria-label="%s" data-name="%s">%s</a>',
+							esc_attr( $plugin['slug'] ),
+							esc_url( $status['url'] ),
+							/* translators: %s: Plugin name and version. */
+							esc_attr( sprintf( _x( 'Install %s now', 'plugin', 'default' ), $name ) ),
+							esc_attr( $name ),
+							esc_html__( 'Install Now', 'default' )
+						);
+					} else {
+						$action_links[] = sprintf(
+							'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+							esc_html( _x( 'Cannot Install', 'plugin', 'default' ) )
+						);
+					}
+				}
+				break;
+
+			case 'update_available':
+				if ( $status['url'] ) {
+					if ( $compatible_php && $compatible_wp ) {
```

Comment:
See above, we may want to check here specifically whether the user has access to update plugins.
```suggestion
					if ( $compatible_php && $compatible_wp && current_user_can( 'update_plugins' ) ) {
```